kclvm/parser/src/session/mod.rs

```diff
@@ -57,9 +57,110 @@ impl ParseSession {
         self.add_parse_err(ParseError::Message {
             message: msg.to_string(),
             span,
+            fix_info: None,
         });
     }
 
+    #[inline]
+    pub fn struct_span_error_with_suggestions(
+        &self,
+        msg: &str,
+        span: Span,
+        suggestion_text: Option<String>,
+        replacement_text: Option<String>,
+    ) {
+        let code_line = self.get_line_content_from_span(span);
+        let modified_code_line =
+            self.generate_modified_code_line(msg, &code_line, replacement_text.as_deref());
+
+        let fix_info = Some(FixInfo {
+            suggestion: suggestion_text,
+            replacement: Some(modified_code_line),
+        });
+
+        self.add_parse_err(ParseError::Message {
+            message: msg.to_string(),
+            span,
+            fix_info,
+        });
+    }
+
+    fn generate_modified_code_line(
+        &self,
+        msg: &str,
+        code_line: &str,
+        replacement_text: Option<&str>,
+    ) -> String {
+        match msg {
```

Comment:
There is a problem with the implementation here. There is no guarantee that the replaced content is the real wrong position. For example

```
if ! "!".is_lower() {}
```
It will be replace to 
```
if not  "not ".is_lower() {}
```